Source/DafnyLanguageServer/Language/DafnyLangParser.cs

```diff
@@ -37,25 +34,25 @@ public sealed class DafnyLangParser : IDafnyParser {
         : new ProgramParser(innerParserLogger, fileSystem);
     }
 
-    public Program Parse(CompilationInput compilation, ErrorReporter reporter, CancellationToken cancellationToken) {
+    public Program Parse(Compilation compilation, CancellationToken cancellationToken) {
```

Comment:
This PR moves the determination of what the root files for a particular compilation are from `ProjectManager` to `Compilation`. Since determining these files can discover user errors, its good to have this be part of Compilation so the handling of those errors works the same as other user errors that occur during compilation, such as parsing errors. Also, `Compilation` will be moved to `DafnyCore` and then the CLI and the server can re-use the same file discovery code.

Since the parser needs to know the root files to parse, which are now stored in the `Compilation`, but not in CompilationInput, it takes a `Compilation` instead of a `CompilationInput`.